game/templates/game/lesson_map.html

```
@@ -270,12 +270,86 @@
 
 <body>
 
+{% if not user.is_authenticated %}
+<div class="alert-banner" id="guest-alert">
+    <div class="alert-content">
+        <span class="alert-icon">💡</span>
+        <span class="alert-text">
+            <a href="{% url 'login' %}" class="alert-link">Sign in</a> or <a href="{% url 'register' %}" class="alert-link">create a free account</a> to save your lesson progress.
+        </span>
+    </div>
+    <button class="alert-close" onclick="dismissAlert()" aria-label="Close message">&times;</button>
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)